miflora/miflora_poller.py

```diff
@@ -258,11 +261,16 @@ def parameter_value(self, parameter, read_cached=True):
     def _check_data(self):
         if self._cache is None:
             return
-        datasum = 0
-        for i in self._cache:
-            datasum += i
-        if datasum == 0:
+        if self._cache[7] > 100: # moisture over 100 procent
+            self._cache = None
+            return
+        if self._firmware_version >= "2.6.6":
+            if sum(self._cache[10:]) == 0:
```

Comment:
I am not sure about the validity of this check.

I do not no the meaning of the data from index 10. But in all the test data I have achieved the values are above 0 when the values are valid. For invalid data the values are often 0.